hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/StoreCache.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.workflows;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.service.mono.token.impl.AccountStore;
+import com.hedera.node.app.state.HederaState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/** A {@code StoreCache} caches stores for all active states. */
+public class StoreCache {
+
+    /**
+     * Returns the {@link AccountStore} for the provided {@link HederaState}.
+     *
+     * @param state the {@code HederaState} that is the base for the {@code AccountStore}
+     * @return the {@code AccountStore} for the provided state, either new or cached
+     */
+    @NonNull
+    public AccountStore getAccountStore(@NonNull final HederaState state) {
+        requireNonNull(state);
+
+        throw new UnsupportedOperationException("not implemented");
+    }
```

Comment:
We need a discussion if its possible to have single `getAccountStore` method accessible for all modules and also for `StoreCache`